pkg/commands/status.go

```diff
@@ -357,34 +331,194 @@ func printPrefix(p string) string {
+	app, appConfig, err := getAppAndApplicationConfiguration(ctx, c, compName, appName, env)
+	if err != nil {
+		return compStatusUnknown, HealthStatusNotDiagnosed, "", err
+	}
+	if app == nil || appConfig == nil {
+		return compStatusUnknown, HealthStatusNotDiagnosed, "", errors.New(ErrNotLoadAppConfig)
+	}
+
+	wlStatus, foundWlStatus := getWorkloadStatusFromAppConfig(appConfig, compName)
+	// make sure component already initilized
+	if !foundWlStatus {
+		appConfigConditionMsg := appConfig.Status.GetCondition(runtimev1alpha1.TypeSynced).Message
+		return compStatusUnknown, HealthStatusUnknown, "", fmt.Errorf(ErrFmtNotInitialized, appConfigConditionMsg)
+	}
+	// check whether referenced a HealthScope
+	var healthScopeName string
+	for _, v := range wlStatus.Scopes {
+		if v.Reference.Kind == kindHealthScope {
+			healthScopeName = v.Reference.Name
+		}
+	}
+	if len(healthScopeName) == 0 {
+		// no health scope referenced
+		statusInfo, _, _, err := getWorkloadInstanceStatusAndCreationTime(ctx, c, env.Namespace, wlStatus.Reference)
+		if err != nil {
+			return compStatusUnknown, HealthStatusUnknown, "", err
+		}
+		return compStatusHealthCheckDone, HealthStatusNotDiagnosed, statusInfo, nil
+	}
+	var healthScope v1alpha2.HealthScope
+	if err = c.Get(ctx, client.ObjectKey{Namespace: env.Namespace, Name: healthScopeName}, &healthScope); err != nil {
+		return compStatusUnknown, HealthStatusUnknown, "", err
+	}
+	var wlhc *v1alpha2.WorkloadHealthCondition
+	for _, v := range healthScope.Status.WorkloadHealthConditions {
+		if v.ComponentName == compName {
+			wlhc = v
+		}
+	}
+	if wlhc == nil {
+		return compStatusUnknown, HealthStatusUnknown, "", fmt.Errorf("cannot get health condition from the health scope: %s", healthScope.Name)
+	}
+	healthStatus := wlhc.HealthStatus
+	if healthStatus == HealthStatusUnknown {
+		healthStatus = HealthStatusNotDiagnosed
+		statusInfo, _, _, err := getWorkloadInstanceStatusAndCreationTime(ctx, c, env.Namespace, wlStatus.Reference)
+		if err != nil {
+			return compStatusUnknown, HealthStatusUnknown, "", errors.Wrap(err, "WARN: The component type is unknown to HealthScope and cannot get status.")
+		}
+		healthInfo := fmt.Sprintf("WARN: The component type is unknown to HealthScope.\nYou may check component status with [%s/%s] status: \n%s",
+			wlhc.TargetWorkload.Kind, wlhc.TargetWorkload.Name, statusInfo)
+		return compStatusHealthCheckDone, healthStatus, healthInfo, nil
+	}
+	if healthStatus == HealthStatusUnhealthy {
+		cTime := appConfig.GetCreationTimestamp()
+		if time.Since(cTime.Time) <= healthCheckBufferTime {
+			return compStatusHealthChecking, HealthStatusUnknown, "", nil
+		}
+	}
+	return compStatusHealthCheckDone, healthStatus, wlhc.Diagnosis, nil
+}
+
+func getAppAndApplicationConfiguration(ctx context.Context, c client.Client, compName, appName string, env *types.EnvMeta) (*application.Application, *v1alpha2.ApplicationConfiguration, error) {
```

Comment:
Normally we consider `App` the same as `ApplicationConfiguration`, so I am a little bit confused about the function name.

Reply:
Yes, a wired name... they are the same logically but still refer to different typed structs.

Reply:
Just naming is `getApp` is good.